src/script/lua_api/l_env.cpp

```diff
@@ -1046,6 +1046,20 @@ int ModApiEnvMod::l_clear_objects(lua_State *L)
 	return 0;
 }
 
+// upgrade_mapblocks(threshold_version)
+// Upgrade all mapblocks below the threshold version
+int ModApiEnvMod::l_upgrade_mapblocks(lua_State *L)
+{
+	GET_ENV_PTR;
+
+	// Range is sanity-checked in ServerEnvironment::upgradeMapBlocks
+	int version = luaL_checkint(L, 1);
+	env->upgradeMapBlocks(version);
```

Comment:
as i see it's done synchronously and can take ages.

I suggest to spawn a thread and do it asynchronously for non loaded blocks.

Also i suggest to implement a command line switch like we have to migrate db in order to upgrade db synchronously at startup, or a minetest.conf flag, as you like. It's clearly better than a command line from the in game admin, which will just break end user experience if playing (except if you succeed to implement it asynchronously)